File: examples/typescript/part-10/36-ai-agent-security-testing-workshop.ts
// Illustrative examples for notes/part-10-course-appendix/36-ai-agent-security-testing-workshop.md
// Not for production use. Licensed under MIT (see LICENSE-CODE).

type Severity = "Critical" | "High" | "Medium" | "Low";

interface Finding {
  id: string;
  title: string;
  severity: Severity;
  area: string;
  expected: string;
  actual: string;
  fix?: string;
  regressionTest?: string;
}

/** Reject incomplete findings; Critical/High require a regression test. */
function validate(f: Finding): void {
  if (!f.id || !f.title) {
    throw new Error("id and title required");
  }
  if (!f.area || !f.severity) {
    throw new Error("area and severity required");
  }
  if (!f.expected || !f.actual) {
    throw new Error("expected and actual required");
  }
  if (
    (f.severity === "Critical" || f.severity === "High") &&
    !f.regressionTest
  ) {
    throw new Error(`${f.severity} finding requires regression test`);
  }
}

export { validate };
export type { Finding, Severity };

File: examples/python/part-10/36-ai-agent-security-testing-workshop.py
# Illustrative examples for notes/part-10-course-appendix/36-ai-agent-security-testing-workshop.md
# Not for production use. Licensed under MIT (see LICENSE-CODE).

from __future__ import annotations

from dataclasses import dataclass


CRITICAL = "Critical"
HIGH = "High"
MEDIUM = "Medium"
LOW = "Low"


@dataclass
class Finding:
    id: str
    title: str
    severity: str
    area: str
    expected: str
    actual: str
    fix: str = ""
    regression_test: str = ""


def validate(f: Finding) -> None:
    """Reject incomplete findings; Critical/High require a regression test."""
    if not f.id or not f.title:
        raise ValueError("id and title required")
    if not f.area or not f.severity:
        raise ValueError("area and severity required")
    if not f.expected or not f.actual:
        raise ValueError("expected and actual required")
    if f.severity in (CRITICAL, HIGH) and not f.regression_test:
        raise ValueError(f"{f.severity} finding requires regression test")
